### PR TITLE
Feature/core fixes

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: include/inviwo/core/processors/processor.h [hunk 1]
```diff
@@ -194,13 +198,17 @@ class IVW_CORE_API Processor : public PropertyOwner,
      */
     void setIdentifier(std::string_view identifier);
     virtual const std::string& getIdentifier() const override;
+    NameDispatcherHandle onIdentifierChange(
```
Comment (martinfalk) [inline include/inviwo/core/processors/processor.h:207]: maybe add comment regarding arguments of std::function. Assuming "prev" and "new" identifier?

File: include/inviwo/core/processors/processor.h [hunk 2]
```diff
@@ -194,13 +198,17 @@ class IVW_CORE_API Processor : public PropertyOwner,
...
+        std::function<void(std::string_view, std::string_view)>);
 
     /**
      * Name of processor, arbitrary string. By default initialized to the ProcessorInfo displayName.
      * This name will be shown on various graphical representations.
      */
     void setDisplayName(std::string_view displayName);
     const std::string& getDisplayName() const;
+    NameDispatcherHandle onDisplayNameChange(
+        std::function<void(std::string_view, std::string_view)>);
```
Comment (martinfalk) [inline include/inviwo/core/processors/processor.h]: comment re std::function

File: modules/qtwidgets/src/processors/processordockwidgetqt.cpp [hunk 3]
```diff
@@ -52,7 +52,7 @@ ProcessorDockWidgetQt::ProcessorDockWidgetQt(Processor* p, const QString& title,
     ivec2 dim = ProcessorWidget::getDimensions();
     ivec2 pos = ProcessorWidget::getPosition();
 
-    setWindowTitle(QString::fromStdString(processor_->getIdentifier()));
+    setWindowTitle(QString::fromStdString(processor_->getDisplayName()));
```
Comment (martinfalk) [inline modules/qtwidgets/src/processors/processordockwidgetqt.cpp]: use `utilqt::fromQString`
Note that the display name is not necessarily unique!

File: modules/qtwidgets/src/tf/tfpropertydialog.cpp [hunk 4]
```diff
@@ -416,6 +390,22 @@ void TFPropertyDialog::showEvent(QShowEvent* event) {
     PropertyEditorWidgetQt::showEvent(event);
 }
 
+void TFPropertyDialog::updateTitleFromProperty() {
+    if (!property_->getOwner()) return;
+
+    auto processorName = property_->getOwner()->getProcessor()->getDisplayName();
+    auto windowTitle =
+        "Transfer Function Editor - " + property_->getDisplayName() + " (" + processorName + ")";
```
Comment (martinfalk) [inline modules/qtwidgets/src/tf/tfpropertydialog.cpp]: use fmt::format?

**Review by martinfalk — CHANGES_REQUESTED** [review]

**Review by martinfalk — APPROVED** [review]

**Review by ResearchDaniel — COMMENTED** [review]
How are the added callbacks removed?

**Comment by petersteneteg** [comment]
you just drop the handle

**Pull request merged**